
version: "3.7"
services:

  nginx:
    image: nginx:1.19.5
    networks:
      - gis-woo-network
    volumes:
      - gis-woo-media:/data/media/
      - gis-woo-static:/data/static/
      - gis_log:/var/log/nginx/
      - /home/ubuntu/nginx.conf:/etc/nginx/nginx.conf
    ports:
      - 80:80

  django:
    image: gis_woo_django:5
    networks:
      - gis-woo-network
    secrets:
      - DJANGO_SECRET_KEY
      - MARIADB_USER
      - MARIADB_PASSWORD
    volumes:
      - gis-woo-media:/home/gis_2ban_2/media/
      - gis-woo-static:/home/gis_2ban_2/staticfiles/

  mariadb:
    image: mariadb:10.5
    networks:
      - gis-woo-network
    volumes:
      - gis_mariadb:/var/lib/mysql
    secrets:
      - MARIADB_USER
      - MARIADB_PASSWORD
      - MARIADB_ROOT_PASSWORD
    environment:
      MARIADB_DATABASE: django
      MARIADB_USER_FILE: /run/secrets/MARIADB_USER
      MARIADB_PASSWORD_FILE: /run/secrets/MARIADB_PASSWORD
      MARIADB_ROOT_PASSWORD_FILE: /run/secrets/MARIADB_ROOT_PASSWORD


networks:
  gis-woo-network:

volumes:
  gis-woo-media:
  gis-woo-static:
  gis_mariadb:
  gis_log:

secrets:
  MARIADB_USER:
    external: true
  MARIADB_PASSWORD:
    external: true
  MARIADB_ROOT_PASSWORD:
    external: true
  DJANGO_SECRET_KEY:
    external: true
